src/sqs/mod.rs

```diff
@@ -0,0 +1,313 @@
+use std::env;
+use std::rc::Rc;
+use std::time::Duration;
+use std::convert::TryFrom;
+
+use rocket::Config;
+use rusoto_core::Region;
+use rusoto_sqs::{DeleteMessageRequest, Message, ReceiveMessageRequest, Sqs, SqsClient};
+use serde_json;
+
+use error::Result;
+use failure::{self, err_msg};
+use logging::RBLogger;
+
+#[derive(Default, Debug, Clone)]
+pub struct SyncEvent {
+    pub event: String, // act on "delete", "device:delete"; ignore others
+    pub uid: String,
+    pub id: String,
+    pub handle: String,
+}
+
+impl TryFrom<Message> for SyncEvent {
+    type Error = failure::Error;
+
+    fn try_from(msg: Message) -> Result<Self> {
+        let body = match msg.body {
```

Comment:
```rust
let body = msg.body.ok_or_else(|| err_msg("Missing body")?;
```

Reply:
FnOnce not implemented for failure::Error. Seems like a reasonable trait to add, though. I'll see about doing that.